test/Feature/UpdateSpec.hs

```diff
@@ -91,23 +100,38 @@ spec = do
           }
 
       it "can update multiple items" $ do
-        replicateM_ 10 $ post "/auto_incrementing_pk"
-          [json| { non_nullable_string: "a" } |]
-        replicateM_ 10 $ post "/auto_incrementing_pk"
-          [json| { non_nullable_string: "b" } |]
-        _ <- request methodPatch
-          "/auto_incrementing_pk?non_nullable_string=eq.a" []
-          [json| { non_nullable_string: "c" } |]
-        g <- get "/auto_incrementing_pk?non_nullable_string=eq.c"
-        liftIO $ simpleHeaders g
-          `shouldSatisfy` matchHeader "Content-Range" "0-9/\\*"
+        get "/no_pk?select=a&b=eq.1"
+          `shouldRespondWith`
+            [json|[]|]
+
+        request methodPatch "/no_pk?b=eq.0"
+            [("Prefer", "tx=commit")]
+            [json| { b: "1" } |]
+          `shouldRespondWith`
+            ""
+            { matchStatus  = 204
+            , matchHeaders = ["Content-Range" <:> "0-1/*"
+                             , "Preference-Applied" <:> "tx=commit" ]
+            }
+
+        -- check it really got updated
+        get "/no_pk?select=a&b=eq.1"
+          `shouldRespondWith`
+            [json|[ { a: "1" }, { a: "2" } ]|]
+
+        -- put value back for other tests
+        request methodPatch "/no_pk?b=eq.1"
+            [("Prefer", "tx=commit")]
+            [json| { b: "0" } |]
+          `shouldRespondWith`
+            204
 
       it "can set a column to NULL" $ do
-        _ <- post "/no_pk" [json| { a: "keepme", b: "nullme" } |]
```

Comment:
Some "preparation" requests were removed, because in those cases existing data could be used - changes are rolled back anyway.